src/uffd_handler/src/prefaulter.rs

```diff
@@ -21,11 +26,266 @@ impl UffdPrefaulter {
     pub fn new(before: usize, after: usize) -> Result<Self, PrefaulterError> {
         Ok(UffdPrefaulter {
             bytes_before: before
-                .checked_add(KIB_MULT)
-                .ok_or_else(|| InvalidAmount(String::from("before"), before))?,
+                .checked_mul(KIB_MULT)
+                .ok_or_else(|| PrefaulterError::InvalidAmount(String::from("before"), before))?,
             bytes_after: after
-                .checked_add(KIB_MULT)
-                .ok_or_else(|| InvalidAmount(String::from("after"), before))?,
+                .checked_mul(KIB_MULT)
+                .ok_or_else(|| PrefaulterError::InvalidAmount(String::from("after"), after))?,
         })
     }
+
+    /// Return the start address of the chunk to prefault into memory.
+    /// When there is no customized prefaulting configured, the minimum chunk to copy to RAM
+    /// is the faulting page.
```

Comment:
Because there are a couple of steps performed by this method, I would enhance the docs comment a little bit like this:

> /// Return the start address of the chunk to prefault into memory. The address is obtained
    /// by subtracting `bytes_before` amount of bytes from the start of the faulting page and
    /// rounding down to a page start. If the computed start address is smaller then the start of
    /// current region, the start of the current memory region is returned. Furthermore, a check for
    /// already existing pages in RAM is performed such that the first Uninitialized page beginning
    /// from the prefaulting start address (that was just computed) is returned eventually.
    /// When there is no customized prefaulting configured, the minimum chunk to copy to RAM
    /// is the faulting page.